lib/kafo/logger.rb

```diff
@@ -13,7 +13,7 @@ def initialize(name = 'root')
 
     def log(level, *args, &block)
       if Logging.buffering?
-        Logging.to_buffer(@name, level, args, &block)
+        Logging.to_buffer(@name, ::Logging::LogEvent.new(@name, ::Logging::LEVELS[level.to_s], *args, false))
```

Comment:
I noticed you dropped the `&block` here, and from what I can tell neither `Logging.to_buffer()` nor `::Logging::LogEvent.new()` has anything to do with such an argument.

It looks like `&block` _can_ be passed to `::Logging::Logger.some_level` ( https://github.com/TwP/logging/blob/18cc826ae65fee63c120a297d3b0235337980058/lib/logging/logger.rb#L112 ) which we support in theory with L19: `@logger.send(level, *args, &block)`

If I understood that correctly it means we have the capability to do things like:

```ruby
logger.notice do
  a = something()
  b = something_else()
  even_more_stuff(a, b)
end
```

which would log an event at level `NOTICE` with `data = even_more_stuff(a, b)`

That's pretty neat, but I looked for places where we actually use it and I couldn't find any

```shell
$ egrep -nr 'log\(|log |logger\.(debug|info|notice|warn|error)' lib/ bin/ test/kafo/logger_test.rb | egrep ' {|&.+\)| do'
lib/kafo/logger.rb:14:    def log(level, *args, &block)
lib/kafo/logger.rb:25:        log(level, *args, &block)
test/kafo/logger_test.rb:23:      it "add log messages to buffer" do
$
```

Reply:
I think IF we had tried to use that pattern `logger.some_level { some_block }` when `logger.buffering?` then before this change `logger.to_buffer` just drops the `&block` anyway, and the same would happen now with the way you have it written.

I think that's fine perhaps -- we already don't actually use this as far as I could tell -- but in that case maybe we should make things simple and drop all this `&block` stuff on L14, L19, L24, and L25 since it would already fail silently some of the time.

Reply:
Or to keep it around we could try to make `Logging::to_buffer` do something with `&block`. Something like...

```ruby
def to_buffer(*args, &block)
  return buffer << yield if block_given?
  buffer << args
end
```
and,

```suggestion
        Logging.to_buffer(@name, ::Logging::LogEvent.new(@name, ::Logging::LEVELS[level.to_s], *args, false), &block)
```

then you could do stuff like (totally made up example to illustrate the idea):

```ruby
basic_checks.each do |check|
  logger.notice(name = check.name) do
    result = "success!" if check.run() rescue "FAIL!"
    "ran basic check: #{name} with result: #{result}"
  end
end
```

no idea if we'd ever have a reason to use something like this where we pass blocks to `logger.level` but I guess we should either support it always, including when `logger.buffering?`, or else not at all.